ci/playbooks/edpm/update.yml

```
@@ -0,0 +1,35 @@
+---
+- name: "Run ci/playbooks/edpm/update.yml"
+  hosts: "{{ cifmw_zuul_target_host | default('all') }}"
+  gather_facts: true
+  tasks:
+    - name: Filter out host if needed
+      when:
+        - cifmw_zuul_target_host is defined
+        - cifmw_zuul_target_host != 'all'
+        - inventory_hostname != cifmw_zuul_target_host
+      ansible.builtin.meta: end_host
+
+    - name: Check for edpm-ansible.yml file
+      ansible.builtin.stat:
+        path: "{{ ansible_user_dir }}/ci-framework-data/artifacts/edpm-ansible.yml"
+      register: edpm_file
+
+    - name: Run Podified EDPM deployment
+      ansible.builtin.command:
+        chdir: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework"
+        cmd: >-
+          ansible-playbook update-edpm.yml
+          -i "{{ ansible_user_dir }}/ci-framework-data/artifacts/zuul_inventory.yml"
+          -e @scenarios/centos-9/base.yml
+          -e @scenarios/centos-9/edpm_ci.yml
+          {%- if edpm_file.stat.exists %}
+          -e @{{ ansible_user_dir }}/ci-framework-data/artifacts/edpm-ansible.yml
+          {%- endif %}
+          {%- if cifmw_extras is defined %}
+          {%-   for extra_var in cifmw_extras %}
+          -e "{{   extra_var }}"
+          {%-   endfor %}
+          {%- endif %}
+          -e "@{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/zuul-params.yml"
+          -e "cifmw_repo_setup_promotion={{ cifmw_repo_setup_promotion_target }}"
```

Can we move `zuul-params.yml` line in last so that vars passed from zuul jobs have higher precedence. Like this
```suggestion
          -e "cifmw_repo_setup_promotion={{ cifmw_repo_setup_promotion_target }}"
          -e "@{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/zuul-params.yml"
         
```

---

No, this is done on purpose. We want to override cifmw_repo_setup_promotion with  cifmw_repo_setup_promotion_target on update run.

---

If we move zuul-params last  -e "cifmw_repo_setup_promotion={{ cifmw_repo_setup_promotion_target }}" has no effect, because last -e param always win.

---

Thank you for explaining it on call. I can mark this resolved.